src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -256,40 +149,43 @@ class Decimal extends React.Component {
    * Format a user defined value
    */
   formatValue = (value) => {
-    invariant(
-      !this.isNaN(value),
-      `The supplied decimal (${value}) is not a number`
-    );
+    if (this.isNaN(value)) {
+      return value;
+    }
+
     if (value === "") {
       return value;
     }
-    return I18nHelper.formatDecimal(value, this.getSafePrecisionProp());
-  };
 
-  /**
-   * Perform a string replacement on the input so we can see what the value will be after the change
-   * This allows us to prevent the user input if it is invalid
-   */
-  replace = (ev) => {
-    const {
-      target: { selectionStart, selectionEnd, value },
-      clipboardData,
-      type,
-    } = ev;
-    let { key: change } = ev;
+    const { separator } = I18nHelper.format();
```

Comment:
I've sent some code on slack to update this with.

Reply:
👍🏻 Applied.